client/my-sites/stats/stats-site-overview/index.jsx

```diff
@@ -2,54 +2,47 @@
  * External dependencies
  */
 import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
-import observe from 'lib/mixins/data-observe';
-import route from 'lib/route';
 import Card from 'components/card';
 import StatsTabs from '../stats-tabs';
 import StatsTab from '../stats-tabs/tab';
 import SectionHeader from 'components/section-header';
+import QuerySiteStats from 'components/data/query-site-stats';
+import { getSiteStatsForQuery } from 'state/stats/lists/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
 
-export default React.createClass( {
-	displayName: 'StatsSiteOverview',
-
+const StatsSiteOverview = React.createClass( {
 	proptypes: {
-		site: PropTypes.object.isRequired,
-		summaryData: PropTypes.object.isRequired,
-		path: PropTypes.string.isRequired
+		siteId: PropTypes.number,
+		siteSlug: PropTypes.string,
+		period: PropTypes.string,
+		date: PropTypes.string,
+		path: PropTypes.string.isRequired,
+		query: PropTypes.object,
+		summaryData: PropTypes.object,
+		insights: PropTypes.bool,
+		title: PropTypes.string
 	},
 
-	mixins: [ observe( 'summaryData' ) ],
-
 	isValueLow( value ) {
 		return ! value || 0 === value;
 	},
 
 	render() {
-		const { site, path, summaryData, insights } = this.props;
-		const { views, visitors, likes, comments } = summaryData.data;
-		const siteStatsPath = [ path, site.slug ].join( '/' );
+		const { siteId, siteSlug, path, summaryData, query, title } = this.props;
+		const { views, visitors, likes, comments } = summaryData;
+		const siteStatsPath = [ path, siteSlug ].join( '/' );
 		let headerPath = siteStatsPath;
-		let title;
-
-		if ( insights ) {
-			title = this.translate( 'Today\'s Stats' );
-		} else {
-			title = site.title;
-			headerPath = route.getStatsDefaultSitePage( site.slug );
-		}
-
-		if ( ! this.props.summaryData ) {
-			return;
-		}
 
 		return (
 			<div>
+				{ siteId && <QuerySiteStats siteId={ siteId } statType="statsSummary" query={ query } /> }
```

Comment:
Found an obscure issue with this.

![0](https://cloud.githubusercontent.com/assets/1779930/19907064/ab59f150-a053-11e6-8f60-721902c8843b.png)

When reloading stats page with a fresh localStorage, the logic is such that when the site is not present, [the `siteId` prop is passed as `0`](https://github.com/Automattic/wp-calypso/blob/d980735450e668cfb6d740829c6399e2cb9823a5/client/my-sites/stats/stats-insights/index.jsx#L85). The `siteId &&` technique can only be used if the falsey variant of the value is `null`, `undefined`, or `false`, otherwise it's rendered as `0` verbatim.

Demo: http://codepen.io/aduth/pen/edqbMB